### ISSUE TITLE
Expected output of QLinearMatMul: using astype or np.clip?

### ISSUE THREAD

**Issue by tungld** [onnx/onnx#6169]
# Ask a Question

### Question
In the test case [qlinearmatmul_int8_float32](https://github.com/onnx/onnx/blob/main/onnx/backend/test/case/node/qlinearmatmul.py#L64C40-L64C68), the output is `[41, -12, -9], [1, -75, 20]]`. Should it be `[41, -12, -9], [1, -75, -128]]` where `20` should be `-128`? 

### Further information

When I looked at [the reference python script for qlinearmatmul](https://github.com/onnx/onnx/blob/main/onnx/reference/ops/op_qlinear_matmul.py), the script to prepare reference ouput for qlinearmatmul:
```python
        A = a.astype(np.int32)
        if a_zero_point is not None:
            A -= a_zero_point.astype(np.int32)
        B = b.astype(np.int32)
        if b_zero_point is not None:
            B -= b_zero_point.astype(np.int32)
        C = np.matmul(A, B)
        D = C * (a_scale * b_scale / y_scale)
        if y_zero_point is not None:
            D += y_zero_point
            return (np.rint(D).astype(y_zero_point.dtype),)
        return (np.rint(D).astype(a.dtype),)
```
If I run the script with the input of `qlinearmatmul_int8_float32`, the value of `np.rint(D)` is
```
[[  41  -12   -9]
 [   1  -75 -236]]
```
The last value `-236` is outside the range of int8, and when calling `astype(y_zero_point.dtype)` (where y_zero_point.dtype is int8), `-236` became `20`:
```
[[  41  -12   -9]
 [   1  -75 20]]
```

My question is whether `20` is the expected result when converting `-236` or it should be `-128` by clipping `-236` into to the range [-128, 127]?
In other words, which is correct: calling `np.rint(D).astype(y_zero_point.dtype)` or calling `np.clip(np.rint(D), -128, 127)`?
